### DISCUSSION TITLE
Add support to duplicated meta tag names

### DISCUSSION THREAD

**Discussion by felquis** [remix-run/remix#4260] [Proposals]
Hello everyone,

As mentioned at this PR https://github.com/remix-run/remix/pull/2598#issuecomment-1255026226 

Open Graph metatags accept duplicated metatags with the same `name` where their order of appearance plays a role.

As described at: https://ogp.me/#array
> If a tag can have multiple values, just put multiple versions of the same <meta> tag on your page. The first tag (from top > to bottom) is given preference during conflicts.
> 
> ```html
> <meta property="og:image" content="https://example.com/rock.jpg" />
> <meta property="og:image" content="https://example.com/rock2.jpg" />
> ```
> Put structured properties after you declare their root tag. Whenever another root element is parsed, that structured property is considered to be done and another one is started.
> 
> For example:
> ```html
> <meta property="og:image" content="https://example.com/rock.jpg" />
> <meta property="og:image:width" content="300" />
> <meta property="og:image:height" content="300" />
> <meta property="og:image" content="https://example.com/rock2.jpg" />
> <meta property="og:image" content="https://example.com/rock3.jpg" />
> <meta property="og:image:height" content="1000" />
> ```
> means there are 3 images on this page, the first image is 300x300, the middle one has unspecified dimensions, and the last one is 1000px tall.

I want to mention this, hopefully it will help you improve the tool.

Thank you.

**Comment by felquis** [comment]
In response to https://github.com/remix-run/remix/pull/2598#issuecomment-1295612581

@chaance thank you for mentioning me.

Can you show me how the usage you showed me can be
applied to reproduce the following usage suggested by Open Graph?
```html
<meta property="og:image" content="https://example.com/rock.jpg" />
<meta property="og:image:width" content="300" />
<meta property="og:image:height" content="300" />
<meta property="og:image" content="https://example.com/rock2.jpg" />
<meta property="og:image" content="https://example.com/rock3.jpg" />
<meta property="og:image:height" content="1000" />
```

It doesn't seem to fit at all, so just wanted to give you this follow up on your answer.

**Reply by chaance** [reply]
@felquis Yeah I don't think structured properties (`og:image:width`) are going to work right unfortunately. We'll need to fix that.

**Reply by machour** [reply]
@felquis an RFC was published for a better `meta` function. Could you give that a read and comment if you see something missing? https://github.com/remix-run/remix/discussions/4462

**Reply by felquis** [reply]
Yes team, if the meta function returns an array of meta tags similar to links it would solve this use case.
But, watch out for meta tags deduplication, since the above example is a legit case used in the market.

Thank you so much for your work <3
